### PR TITLE
add pagination to dynamo scan

### PR DESCRIPTION
**Pull request by jinyangli**
I only changed Scan to support pagination and not ParallelScan....

### CODE REVIEW

File: dynamodb/scan.go [hunk 1]
```diff
@@ -6,46 +6,63 @@ import (
...
+			return nil, lastKey, errors.New(message)
 		}
 		results[i] = parseAttributes(item)
 	}
-	return results, nil
+	return results, lastKey, nil
 
 }
 
-func (t *Table) Scan(attributeComparisons []AttributeComparison) ([]map[string]*Attribute, error) {
```
Comment (aalness) [inline dynamodb/scan.go]: Maybe we should keep a version that doesn't take a start and add the new method as `ScanWithStart` so we can ignore upstream tests.

**Comment by aalness** [comment]
Looks fine except for the one comment. We also don't want to break old code in case someone is depending on this interface.

**Comment by jinyangli** [comment]
Done. Thanks for the suggestion!

**Comment by jinyangli** [comment]
Done. Thanks for the suggestion!

**Pull request merged**